### PR TITLE
_pixelbuf support

### PR DESCRIPTION
**Pull request by rhooper**
This is a WIP because pypixelbuf isn't done yet, and the _pixelbuf API changes haven't been merged.  Looking for feedback!

### CODE REVIEW

**Comment by tannewt** [comment]
Nice! This looks good to me.

I commented on the NeoPixel SPI PR (https://github.com/adafruit/Adafruit_CircuitPython_NeoPixel/pull/53) about removing it.

**Comment by rhooper** [comment]
Ready for review.  This might need doc changes that discuss pixelbuf and/or pypixelbuf.

Should be good for freezing into boards.  Should we merge this into a branch other than maste first?

**Comment by rhooper** [comment]
Boards that reference NeoPixel:
./ports/atmel-samd/boards/feather_m0_express_crickit/mpconfigboard.mk
./ports/atmel-samd/boards/circuitplayground_express_displayio/mpconfigboard.mk
./ports/atmel-samd/boards/circuitplayground_express_crickit/mpconfigboard.mk
./ports/atmel-samd/boards/circuitplayground_express/mpconfigboard.mk
./ports/atmel-samd/boards/hallowing_m0_express/mpconfigboard.mk
./ports/atmel-samd/boards/robohatmm1_m4/mpconfigboard.mk

File: docs/conf.py [hunk 1]
```diff
@@ -15,7 +15,7 @@
     'sphinx.ext.viewcode',
 ]
 
-autodoc_mock_imports = ["neopixel_write"]
+autodoc_mock_imports = ["neopixel_write", "pypixelbuf"]
```
Comment (tannewt) [inline docs/conf.py]: Is this needed still? I would think the pypi package would work.

File: .pylintrc [hunk 2]
```diff
@@ -119,7 +119,8 @@ spelling-store-unknown-words=no
 [MISCELLANEOUS]
 
 # List of note tags to take in consideration, separated by a comma.
-notes=FIXME,XXX,TODO
+# notes=FIXME,XXX,TODO
+notes=FIXME,XXX
```
Comment (siddacious) [inline .pylintrc]: Can we revert this change? It doesn't look like you're adding any TODO comments and we're trying to avoid diffs from the cookiecutter generated code where possible.

As a fortuitous second effect, it should trigger the now merged actions build trigger to fire so we can check that this PR will work ok with the new process

Comment (rhooper) [reply .pylintrc]: Reverted.
